x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeScriptFieldMapperTests.java

```diff
@@ -145,6 +148,51 @@ public void testDate() throws IOException {
         assertEquals(Strings.toString(mapping("date")), Strings.toString(mapperService.documentMapper()));
     }
 
+    public void testDateWithFormat() throws IOException {
+        CheckedSupplier<XContentBuilder, IOException> mapping = () -> mapping("date", b -> b.field("format", "yyyy-MM-dd"));
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping.get()).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(RuntimeScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping.get()), Strings.toString(mapperService.documentMapper()));
+    }
+
+    public void testDateWithLocale() throws IOException {
+        CheckedSupplier<XContentBuilder, IOException> mapping = () -> mapping("date", b -> b.field("locale", "en_GB"));
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping.get()).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(RuntimeScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping.get()), Strings.toString(mapperService.documentMapper()));
+    }
+
+    public void testDateWithLocaleAndFormat() throws IOException {
+        CheckedSupplier<XContentBuilder, IOException> mapping = () -> mapping(
+            "date",
+            b -> b.field("format", "yyyy-MM-dd").field("locale", "en_GB")
+        );
+        MapperService mapperService = createIndex("test", Settings.EMPTY, mapping.get()).mapperService();
+        FieldMapper mapper = (FieldMapper) mapperService.documentMapper().mappers().getMapper("field");
+        assertThat(mapper, instanceOf(RuntimeScriptFieldMapper.class));
+        assertEquals(Strings.toString(mapping.get()), Strings.toString(mapperService.documentMapper()));
+    }
+
+    public void testNonDateWithFormat() throws IOException {
+        String runtimeType = randomValueOtherThan("date", () -> randomFrom(runtimeTypes));
+        Exception e = expectThrows(
+            MapperParsingException.class,
+            () -> createIndex("test", Settings.EMPTY, mapping(runtimeType, b -> b.field("format", "yyyy-MM-dd")))
+        );
+        assertThat(e.getMessage(), equalTo("Failed to parse mapping: format not supported by runtime_type [" + runtimeType + "]"));
```

Comment:
nit: I think that looking at this error message, there is a slight change that it makes users think that the specific value they provided for the format parameter is not supported, more than providing any format. same for locale. Maybe we can rephrase it?